src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -284,6 +288,53 @@ private static Overload isIp() {
         null);
   }
 
+  /**
+   * Creates a custom function overload for the "isIpPrefix" operation.
+   *
+   * @return The {@link Overload} instance for the "isIpPrefix" operation.
+   */
+  private static Overload isIpPrefix() {
+    return Overload.overload(
+        OVERLOAD_IS_IP_PREFIX,
+        null,
+        value -> {
+          if (value.type().typeEnum() != TypeEnum.String && value.type().typeEnum() != TypeEnum.Bool ) {
+            return Err.noSuchOverload(value, OVERLOAD_IS_IP_PREFIX, null);
+          }
+          String prefix = (String) value.value();
+          if (prefix.isEmpty()) {
+            return BoolT.False;
+          }
+          return Types.boolOf(validateIPPrefix(prefix, 0L, false));
+        },
+        (lhs, rhs) -> {
+          if (lhs.type().typeEnum() != TypeEnum.String || (rhs.type().typeEnum() != TypeEnum.Int && rhs.type().typeEnum() != TypeEnum.Bool)) {
+            return Err.noSuchOverload(lhs, OVERLOAD_IS_IP_PREFIX, rhs);
+          }
+          String prefix = (String) lhs.value();
+          if (prefix.isEmpty()) {
+            return BoolT.False;
+          }
+          if (rhs.type().typeEnum() == TypeEnum.Int) {
+            return Types.boolOf(validateIPPrefix(prefix, rhs.intValue(), false));
+          } else if (rhs.type().typeEnum() == TypeEnum.Bool) {
+            return Types.boolOf(validateIPPrefix(prefix, 0L, rhs.booleanValue()));
+          }
+          return BoolT.False;
```

Comment:
nit: Since we've already verified above the rhs is either Int/Bool, we can remove the `else if` and just `return Types.boolOf(validateIPPrefix(prefix, 0L, rhs.booleanValue()));`